File: LC_345.py
class Solution:
    def reverseVowels(self, st: str) -> str:
        vowels = {'a', 'i', 'o', 'u', 'e', 'A', 'E', 'I', 'O', 'U'}
        s = list(st)
        start, end = 0, len(s)-1
        #icecream

        while start < end:
            if s[start] not in vowels:
                start += 1
                continue
            if s[end] not in vowels:
                end -= 1
                continue

            temp = s[start]
            s[start] = s[end]
            s[end] = temp
            start += 1
            end -= 1


        return "".join(s)
    
print(Solution().reverseVowels("IceCreAm"))

File: LC_345.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public String reverseVowels(String s) {
        
        Set<Character> vow = new HashSet<Character>(Set.of('a', 'e', 'i', 'o', 'u'));
        StringBuilder sb = new StringBuilder(s);
        int start = 0, end = s.length()-1;

        while(start < end) {
            
            if(!vow.contains(Character.toLowerCase(sb.charAt(start)))) {
                start += 1;
                continue;
            }

            if(!vow.contains(Character.toLowerCase(sb.charAt(end)))) {
                end -= 1;
                continue;
            }

            var temp = sb.charAt(start);
            sb.setCharAt(start, sb.charAt(end));
            sb.setCharAt(end, temp);

            start += 1;
            end -= 1;

        }

        return sb.toString();
    }
}